src/large_pages/node_large_page.cc

```diff
@@ -339,20 +339,23 @@ __attribute__((__noinline__))
 MoveTextRegionToLargePages(const text_region& r) {
   void* nmem = nullptr;
   void* tmem = nullptr;
-  int ret = 0;
-
-  size_t size = r.to - r.from;
   void* start = r.from;
+  size_t size = r.to - r.from;
+
+  auto free_mems = OnScopeLeave([&nmem, &tmem, size]() {
+    if (nmem != nullptr && nmem != MAP_FAILED && munmap(nmem, size) == -1)
+      PrintSystemError(errno);
+    if (tmem != nullptr && tmem != MAP_FAILED && munmap(tmem, size) == -1)
+      PrintSystemError(errno);
```

Comment:
```suggestion
    if ((nmem != nullptr && nmem != MAP_FAILED && munmap(nmem, size) == -1) ||
        (tmem != nullptr && tmem != MAP_FAILED && munmap(tmem, size) == -1)) {
      PrintSystemError(errno);
    }
```

Alternatively, it may be a bit cleaner to use a custom smart pointer with a deleter for these...

e.g. bit more verbose...

```c++
  struct Mmap {
    void* mem = nullptr;
    size_t size = 0;
    Mmap(void* start, size_t size, int prot, int flags) {
      mem = mmap(start, size, prot, flags, -1, 0);
    }
  };
  struct MmapDeleter {
    void operator()(Mmap* ptr) const noexcept {
      if (ptr.mem != nullptr &&
          ptr.mem != MAP_FAILED &&
          munmap(ptr.mem, ptr.size) == -1)
        PrintSystemError(errno); 
    }
  };
  using MmapPointer = std::unique_ptr<Mmap, MmapDeleter>;
```

then...

```c++
  MmapPointer nmem;
  MmapPointer tmem;

  // ...
  size_t size = r.to - r.from; 

  // ...
  nmem = std::make_unique(nullptr, size,
                          PROT_READ | PROT_WRITE,
                          MAP_PRIVATE | MAP_ANONYMOUS);
  if (nmem.mem == MAP_FAILED)
    goto fail;

  // ...
  tmem = std::make_unique(start, size,
                          PROT_READ | PROT_WRITE,
                          MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED);
  if (tmem.mem == MAP_FAILED)
    goto fail;
```